### ISSUE TITLE
[GOT-77] [GOT-28] Refactor the code to simplify the database access

### ISSUE THREAD

**Issue by brunocalza** [tablelandnetwork/go-tableland#502]
<sub>From [SyncLinear.com](https://synclinear.com) | [GOT-28](https://linear.app/tableland/issue/GOT-28/refactor-the-code-to-simplify-the-database-access)</sub>

<sub>[GOT-77](https://linear.app/tableland/issue/GOT-77/[got-28]-refactor-the-code-to-simplify-the-database-access)</sub>

**Comment by brunocalza** [comment]
yeah, definitely

**Comment by brunocalza** [comment]
@brunocalza should we push this to ship 2b?

**Comment by brunocalza** [comment]
@avichal opened this for review. it's quite big 😅 so take your time. make sure you read [https://www.notion.so/textile/Database-Access-Code-Refactoring-3bf22e604f7249f894dfb5e153c10dc6](https://www.notion.so/textile/Database-Access-Code-Refactoring-3bf22e604f7249f894dfb5e153c10dc6) and use it as a guide

**Comment by sanderpick** [comment]
seems like there is some double syncing of linear issues. @brunocalza curious if you know what flow led to this?

**Comment by brunocalza** [comment]
yeah, i added a public label to [https://linear.app/tableland/issue/GOT-28/refactor-the-code-to-simplify-the-database-access](https://linear.app/tableland/issue/GOT-28/refactor-the-code-to-simplify-the-database-access) and it created an issue in GH, but looks like that GH issue triggered the creation of this duplicated ticket

**Comment by sanderpick** [comment]
hmm, that's annoying. I've seen it do that when there was a closed issue of the same name in the repo, but I doubt that's what happened here. I'll have to look into it.

**Comment by sanderpick** [comment]
@brunocalza I know you said you wanted to test this some more. All good for release on Tuesday?

**Comment by brunocalza** [comment]
I haven't had the time yet to execute the tests that I wanted. So probably the next one

**Comment by brunocalza** [comment]
@avichal i did an additional commit on this [https://github.com/tablelandnetwork/go-tableland/pull/501/commits/97de5becb42d22c442fe6320d49f912a861f844e](https://github.com/tablelandnetwork/go-tableland/pull/501/commits/97de5becb42d22c442fe6320d49f912a861f844e) would you mind taking a look?

I've abandoned the idea of having two databases (one for reads and one for writes). Having a database with only one connection at time is too problematic. That was an attempt to avoid SQLite `database is locked` error by locking the database access at a higher-level, but that opens the door to a bunch of other problems. It's much better (IMO) to allow multiple db connections and let SQLite handle the locking. SQLIte it's much smarter at handling locks than having one open connection at time. That means we will still get `database is locked` error from time to time because there are multiple threads accessing the SQLite file, but the places that that happens those operations are retryable, so not a big deal. 

In short, this commit essentially removes one of the databases and use only one. And also, does some adjustments in the dashboard.

**Comment by brunocalza** [comment]
It was something we were discussing 2 weeks ago. Some components (in this case it was the nonce tracker) were getting stuck and having a timeout because it was not able to get access to the database (because of the sql.DB lock).
